src/parameterizations/vertical/MOM_diabatic_driver.F90

```diff
@@ -251,7 +251,7 @@ module MOM_diabatic_driver
   real, allocatable, dimension(:,:,:) :: KPP_buoy_flux  !< KPP forcing buoyancy flux [L2 T-3 ~> m2 s-3]
   real, allocatable, dimension(:,:)   :: KPP_temp_flux  !< KPP effective temperature flux
                                                         !! [degC H T-1 ~> degC m s-1 or degC kg m-2 s-1]
-  real, allocatable, dimension(:,:)   :: KPP_salt_flux  !< KPP effective salt flux
+  real, pointer,     dimension(:,:) :: KPP_salt_flux => NULL() !< KPP effective salt flux
```

Comment:
Is there a reason to convert this array to a pointer? GFDL has recently replaced many pointer declarations in MOM6 as either allocatables or locals for improving the performance: https://github.com/NOAA-GFDL/MOM6/pull/5
So, unless there is a limitation for using an allocable array herer, I suggest keeping it an allocatable array

Reply:
The salt flux is also needed in the pseudosalt module, and so I added a pointer for it to [`forcing_type`](https://github.com/NCAR/MOM6/blob/3e4f827d25f3b2c9e9be514e7aa2b14ddee9b7a4/src/core/MOM_forcing_type.F90#L117) and then the diabatic driver ensures the `forcing_type` pointer [points to this field](https://github.com/NCAR/MOM6/blob/3e4f827d25f3b2c9e9be514e7aa2b14ddee9b7a4/src/parameterizations/vertical/MOM_diabatic_driver.F90#L757).

Reply:
Thanks. Now that makes me wondering why `KPP_salt_flux` is a part of `diabatic_CS`. If it was a member of `KPP_CSp` instead, then you wouldn't need to make it a pointer or add it to `forcing_type`. The same goes for `KPP_NLTheat` and other `KPP_*` data above. Why are they not a part of the KPP module? Anyway, these can be deemed outside the scope of this PR, but `KPP_salt_flux` being a part of `KPP_CSp` would simplify this PR a bit.

Reply:
> Thanks. Now that makes me wondering why `KPP_salt_flux` is a part of `diabatic_CS`. If it was a member of `KPP_CSp` instead, then you wouldn't need to make it a pointer or add it to `forcing_type`. The same goes for `KPP_NLTheat` and other `KPP_*` data above. Why are they not a part of the KPP module? Anyway, these can be deemed outside the scope of this PR, but `KPP_salt_flux` being a part of `KPP_CSp` would simplify this PR a bit.

I think this would be an interesting discussion for one of the MOM meetings with GFDL, or maybe via an issue ticket on the mom-ocean repo? I agree that refactoring this in a way that would let us pass the data to pseudosalt without the addition to `forcing_type` would be an improvement over what is in this PR